src/transformers/trainer.py

```diff
@@ -3184,7 +3184,8 @@ def evaluation_loop(
 
             if labels is not None:
                 labels = self.accelerator.gather_for_metrics((labels))
-                labels_host = labels if labels_host is None else nested_concat(labels_host, labels, padding_index=-100)
+                labels_host = [labels] if labels_host is None else labels_host+[labels]
```

Comment:
Why are we no longer using nested_concat here?

Reply:
Because of https://github.com/huggingface/transformers/issues/25939#issuecomment-1704588237
This is a botch to make my evaluation work and keep a list (of batches) of lists (of examples) separated.
As it is now they're merged by `nested_concat` in a way that is irreversible otherwise.
It is likely that the solution should be revisiting `nested_concat`

Reply:
run with a gpu; looks at last cell's output
https://colab.research.google.com/drive/17VzhS0rlswL-zmd1wPZXSRfFUsGDLfWB?usp=sharing

Reply:
If your solution works for your problem, we definitely should consider it. But we need to make sure this doesn't considerably break things existing (see the current failing tests) :)

Reply:
> If your solution works for your problem, we definitely should consider it. But we need to make sure this doesn't considerably break things existing (see the current failing tests) :)

I definitely agree! I am not sure what your preferences are at huggingface.
it is quite confusing the entire way it handles input to the metrics, it seems like you're handling the entire model output into the metric with some intention in mind, tuplizing it and gathering batches etc etc. I am not sure what to suggest as an actual alternative at this time because of that.
it's just impossible to calculate any metric with that detr output as it is now...